kratos/utilities/geometry_utilities.h

```diff
@@ -400,6 +400,47 @@ class GeometryUtils
 
     }
 
+    /**
+     * Calculate the exact distances to the plane interface defined by a set 
+     * of initial distances.
+     * @param ThisGeometry Geometry can be either a triangle or a tetrahedra
+     * @param Distances The distances which define the isosurface as input.
+     * Same argument is used to give the calculated exact distances back
+     */
+    template<std::size_t TSize>
+    static void CalculateExactDistancesToPlane(
+        Element::GeometryType& rThisGeometry,
+        array_1d<double, TSize>& rDistances)
+    {
+        array_1d<Point, TSize> intersection_points;
+        int number_of_intersection_points = CalculateTetrahedraIntersectionPoints(rThisGeometry, rDistances, intersection_points);
+
+		if(number_of_intersection_points == 0) {
+			KRATOS_WARNING("GeometryUtilities") << "Warning: The intersection with interface hasn't found! The distances are" << rDistances << std::endl;
+		}
+        else {
+            BoundedMatrix<double,TSize,TSize-1> DN_DX;
+            array_1d<double, TSize> N;
+            double volume;
+            GeometryUtils::CalculateGeometryData(rThisGeometry, DN_DX, N, volume);
+            array_1d<double, TSize-1> distance_gradient = prod(trans(DN_DX), rDistances);
+            double distance_gradient_norm = norm_2(distance_gradient);
+            if (distance_gradient_norm < 1e-15) distance_gradient_norm = 1e-15; //avoid division by zero
+            distance_gradient /= distance_gradient_norm;
+            //We use the first intersection point as reference
+            const auto ref_point = intersection_points[0].Coordinates();
```

Comment:
why a copy?
```suggestion
            const auto& r_ref_point = intersection_points[0].Coordinates();
```

Reply:
Because I forgot the ampersand :sweat_smile: . Thanks for noticing!